sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityCredential.java

```diff
@@ -35,7 +39,8 @@
             .identityClientOptions(identityClientOptions)
             .build();
         Configuration configuration = Configuration.getGlobalConfiguration().clone();
-        if (configuration.contains(Configuration.PROPERTY_MSI_ENDPOINT)) {
+        if (configuration.contains(Configuration.PROPERTY_MSI_ENDPOINT)
```

Comment:
This would be true in Cloud Shell, and `AppServiceMsiCredential` would fail there. Is Cloud Shell managed identity not supported? It will also be true for Azure Arc, if I remember that design correctly.

Reply:
`AppServiceMsiCredential` works on CloudShell (https://github.com/Azure/azure-sdk/issues/795#issuecomment-559900596) - @g2vinay have we tested Azure Arc?

Reply:
Azure Arc is for the next month release, will test this month.

Reply:
it works in cloud shell, as @jianghaolu  mentioned above.
I have updated the conditional here and it won't be true for Azure Arc anymore.

Azure arc support is yet to be added in all languages.